src/components/BaseGameView.tsx

```diff
@@ -8,31 +8,25 @@ import { Energy } from 'src/components/Energy';
 import { Card } from 'src/GamePlay/Card';
 import { Hand } from 'src/components/Hand';
 
-export interface BaseGameViewProps {
+export interface BaseGameViewState {
     hp: number;
     maxHp: number;
     energy: number;
     maxEnergy: number;
     effectList: StatusEffect[];
     hand: Card[];
+}
+
+export interface BaseGameViewDispatch {
     playCard: (card: Card, sourceId: string, targetId: string) => void;
     endTurn: () => void;
 }
 
-export interface BaseGameViewDispatch extends Pick<BaseGameViewProps,
-"playCard" | "endTurn">{}
-
-export interface BaseGameViewState extends Pick<BaseGameViewProps,
-"hp" |
-"maxHp" |
-"energy" |
-"maxEnergy" |
-"effectList" |
-"hand">{}
+type BaseGameViewProps  = BaseGameViewState & BaseGameViewDispatch
```

Comment:
Sure, we can go this route and see where it lands us.

IIRC, the reason we went at this the other way around is because it decoupled it from redux more. With this approach, you are defining props that come from a shape that is, by definition, the "dispatch" props. If we're attempting to decouple the UI component from redux/connect, it seems kind of counterproductive to define the props that way.

But I mean heck, this is our side project and I think it's reasonable to say that it's unlikely that we will move away from redux, so if we think this is less typing or/and is more aligned with how we think about things, then I think it's a totally reasonable trade-off.